pkg/provider/azure_vmss.go

```diff
@@ -74,9 +74,13 @@ type ScaleSet struct {
 	// this also allows for clusters with both VM and VMSS nodes.
 	availabilitySet VMSet
 
-	vmssCache                 *azcache.TimedCache
-	vmssVMCache               *sync.Map // [resourcegroup/vmssname]*azcache.TimedCache
-	availabilitySetNodesCache *azcache.TimedCache
+	flexScaleSet VMSet
```

Comment:
could you add a comment when flexScaleSet would be used?

Reply:
Done